starter-code/app.js

```diff
@@ -14,24 +14,88 @@ const siteController = require("./routes/siteController");
 // Mongoose configuration
 mongoose.connect("mongodb://localhost/ibi-ironhack");
 
+//require the user model
+const User = require("./models/user");
+const session = require("express-session");
+const bcryptjs = require("bcryptjs");
+const passport = require("passport");
+const LocalStrategy = require("passport-local").Strategy;
+const flash = require("connect-flash");
+
+//enable sessions here
+app.use(
+  session({
+    secret: "our-passport-local-strategy-app",
+    resave: true,
+    saveUninitialized: true
+  })
+);
+app.use(flash());
+
+//initialize passport and session here
+passport.serializeUser((user, cb) => {
+  cb(null, user._id);
+});
+
+passport.deserializeUser((id, cb) => {
+  User.findOne({ _id: id }, (err, user) => {
+    if (err) {
+      return cb(err);
+    }
+    cb(null, user);
+  });
+});
+
+app.use(passport.initialize());
+app.use(passport.session());
+
 // view engine setup
-app.set('views', path.join(__dirname, 'views'));
-app.set('view engine', 'ejs');
+app.set("views", path.join(__dirname, "views"));
+app.set("view engine", "ejs");
 
 // uncomment after placing your favicon in /public
 //app.use(favicon(path.join(__dirname, 'public', 'favicon.ico')));
-app.use(logger('combined'));
+app.use(logger("combined"));
 app.use(bodyParser.json());
 app.use(bodyParser.urlencoded({ extended: false }));
 app.use(cookieParser());
-app.use(express.static(path.join(__dirname, 'public')));
+app.use(express.static(path.join(__dirname, "public")));
 
 // Routes
 app.use("/", siteController);
```

Comment:
The `use` of a controller/router always comes at the end of the pipeline (right before the error handling).  Otherwise, your middleware configuration, such as for passport, will not be relevant to your navigation.